### ISSUE TITLE
set_members throws error on topology change

### ISSUE THREAD

**Issue by SvenW** [derekkraan/horde#78]
So I connect two nodes with `set_members` which results in the following on both nodes:
```[
   {App.DistributedSupervisor, :"node1@127.0.0.1"},
   {App.DistributedSupervisor, :"node2@127.0.0.1"}
]
[
   {App.ProcessRegistry, :"node1@127.0.0.1"},
   {App.ProcessRegistry, :"node2@127.0.0.1"}
]```
I start a process and list all processes in the registry on both nodes which gives me:
```iex(node1@127.0.0.1)3> Horde.Registry.processes(App.ProcessRegistry)
%{
  "process-id" => {#PID<0.674.0>,
   nil}
}

iex(node2@127.0.0.1)3> Horde.Registry.processes(App.ProcessRegistry)
%{
  "process-id" => {#PID<23104.674.0>,
   nil}
}```
When I then use `set_members` again for `DistributedSupervisor` it works on the first node, however the second node raises this exception:
```iex(node2@127.0.0.1)4> Horde.Cluster.set_members(App.DistributedSupervisor, [App.DistributedSupervisor])
13:05:56.638 [debug] Found 1 processes on dead nodes [ pid=<0.420.0> line=500 function=processes_on_dead_nodes/1 module=Horde.SupervisorImpl file=lib/horde/supervisor_impl.ex application=horde ]
** (exit) exited in: GenServer.call(App.DistributedSupervisor, {:set_members, [App.DistributedSupervisor]}, 5000)
    ** (EXIT) an exception was raised:
        ** (ArithmeticError) bad argument in arithmetic expression
            :erlang.rem(1766195506, 0)
            (horde) lib/horde/uniform_distribution.ex:17: Horde.UniformDistribution.choose_node/2
            (horde) lib/horde/supervisor_impl.ex:475: anonymous fn/3 in Horde.SupervisorImpl.handle_topology_changes/1
            (elixir) lib/enum.ex:2934: Enum.filter_list/2
            (horde) lib/horde/supervisor_impl.ex:474: Horde.SupervisorImpl.handle_topology_changes/1
            (horde) lib/horde/supervisor_impl.ex:395: Horde.SupervisorImpl.set_members/2
            (horde) lib/horde/supervisor_impl.ex:102: Horde.SupervisorImpl.handle_call/3
            (stdlib) gen_server.erl:661: :gen_server.try_handle_call/4
    (elixir) lib/gen_server.ex:989: GenServer.call/3```

**Comment by QuantamHD** [comment]
I've been seeing this as well in our Kubernetes deployment

**Comment by derekkraan** [comment]
This was fixed in 36b4fdfe4d0236941ded277688a34f623e224ff5
